.github/workflows/unit-tests.yaml

```diff
@@ -0,0 +1,88 @@
+name: Tests
+on:
+  push:
+    tags:
+    - '*'
+    branches:
+    - master
+    - release-*
+  pull_request:
+
+jobs:
+  go:
+    strategy:
+      matrix:
+        gotags: [ 'GOTAGS=""', 'GOTAGS=release' ]
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/stackrox-io/apollo-ci:stackrox-test-0.3.50
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - name: Cache Go dependencies
+      uses: ./.github/actions/cache-go-dependencies
+
+    - name: Go Unit Tests
+      run: ${{ matrix.gotags }} make go-unit-tests
+  go-postgres:
+    strategy:
+      matrix:
+        gotags: [ 'GOTAGS=""', 'GOTAGS=release' ]
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/stackrox-io/apollo-ci:stackrox-test-0.3.50
+    services:
+      # Label used to access the service container
+      postgres:
+        # Docker Hub image
+        image: postgres
+        # Provide the password for postgres
+        env:
+          POSTGRES_PASSWORD: postgres
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
I'm not sure such `ref` will work for non-pull_request events. Maybe just omit it?
[Documentation](https://github.com/actions/checkout) says:

```yaml
    # The branch, tag or SHA to checkout. When checking out the repository that
    # triggered a workflow, this defaults to the reference or SHA for that event.
    # Otherwise, uses the default branch.
    ref: ''
```

Reply:
https://github.com/stackrox/stackrox/pull/4213

Reply:
It looks like there is a valid usecase for it.
Fetch 0 is used to download whole history so `make tag` will work and `ref` is used to build against commit instead of a merge https://github.com/actions/checkout#checkout-pull-request-head-commit-instead-of-merge-commit

Reply:
Found answer to the Q: will `ref: ${{ github.event.pull_request.head.sha }}` work on non-pull requests?
A: from [here](https://docs.github.com/en/actions/learn-github-actions/contexts#about-contexts)

> If you attempt to dereference a non-existent property, it will evaluate to an empty string.